### DISCUSSION TITLE
Schedule missed

### DISCUSSION THREAD

**Discussion by notAutomne** [apache/airflow#38503] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### If "Other Airflow 2 version" selected, which one?

2.8.1

### What happened?

On a DAG with the following crontab : `"0 6 * * 1-5"`, the DAG didn't run on Monday 25th : 
![image](https://github.com/apache/airflow/assets/2185992/d152dd56-e417-4d18-a54e-32dd963612e2)

### How to reproduce

Here the DAG : 
```PY
from datetime import timedelta

from airflow import DAG
from airflow.decorators import task
from pendulum import now as pendulum_tz_now

from vicatflow.vars import AD_VAULT_PATH, WINRM_SVC_ACCOUNT
from vicatflow.tasks.services.servicemgmt import WSServiceSession, ServiceStartType


_ORIG_ID = "service_PRD_FR_XYZ_SvcXXX_on"
with DAG(
    dag_id=f"{_ORIG_ID}",
    start_date=pendulum_tz_now("Europe/Paris").subtract(days=1),
    schedule="0 6 * * 1-5",
    catchup=False,
    dagrun_timeout=timedelta(minutes=180),
    tags=["prd", "application", "service", "XXX", "france", "staging airflow"],
) as dag:
```

### Operating System

Amazon Linux 2023

### Versions of Apache Airflow Providers

```
apache-airflow-providers-amazon==8.16.0
apache-airflow-providers-celery==3.5.1
apache-airflow-providers-cncf-kubernetes==7.13.0
apache-airflow-providers-common-io==1.2.0
apache-airflow-providers-common-sql==1.10.0
apache-airflow-providers-ftp==3.7.0
apache-airflow-providers-hashicorp==3.6.1
apache-airflow-providers-http==4.8.0
apache-airflow-providers-imap==3.5.0
apache-airflow-providers-microsoft-azure==8.5.1
apache-airflow-providers-microsoft-mssql==3.6.0
apache-airflow-providers-microsoft-psrp==2.5.0
apache-airflow-providers-microsoft-winrm==3.4.0
apache-airflow-providers-postgres==5.10.0
apache-airflow-providers-redis==3.6.0
apache-airflow-providers-sftp==4.8.1
apache-airflow-providers-snowflake==5.2.1
apache-airflow-providers-sqlite==3.7.0
apache-airflow-providers-ssh==3.10.0
```

### Deployment

Other

### Deployment details

Manual deployment on EC2 instances

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment]
Until there are some details that allow to diagnose it, it's at most a discussion. There is no indication or way for anyone to diagnose the issue so if you want to help those who might want to help you to understand your problem, digging in log around the issue and providing them might help someone to decide to spend their free volunteer time to help you, so it's mostly up to you to provide as much information as you can in the hopes of getting help (you have to remember it's a forum where people help each other - it's not a helpdesk).
